edb/server/server.py

```diff
@@ -847,6 +849,14 @@ async def start(self):
             }
             print(f'\nEDGEDB_SERVER_DATA:{json.dumps(ri)}\n', flush=True)
 
+        if self._status_sink is not None:
+            status = {
+                "port": self._listen_port,
+                "socket_dir": str(self._runstate_dir),
+                "main_pid": os.getpid(),
+            }
+            self._status_sink(f'READY={json.dumps(status)}')
```

Comment:
Let's use JSON all the way:

```suggestion
            self._status_sink(json.dumps({"status": "ready", **status}))
```

Reply:
The idea here is that `READY=` is the message type and JSON is payload.  It's also easier to match.

Reply:
The only reason we had prefix with `--echo` is because stdout is full of garbage and we needed a marker. Here though, we can just readlines and parse JSON. So I'm -1 on the `READY=` prefix.

Reply:
Markers are still needed, because `READY` wouldn't necessarily be the only kind of status message in there.  I'm not sure why line-oriented `<msg-kind>=<payload>` protocol is controversial here.  Pure JSON would require full parsing, which wouldn't be possible in shell without an external tool like `jq`.

(FWIW, I modeled this after systemd's `sd_notify`.)

Reply:
>  I'm not sure why line-oriented <msg-kind>=<payload> protocol is controversial here.

Because you're forcing me to parse this primitive protocol. Instead of just reading the line and parsing JSON, I have one extra step of splitting by the first `=`. There's no reason for this.

How about we replace `"status":"ready"` with `"kind":`"ready-message"`?

Reply:
> Because you're forcing me to parse this primitive protocol.

Is this really a problem? `<IDENT>=` is basically message framing, which allows matching of or skipping over messages without parsing the payload.  Hiding the message type in the payload is not a better design in this regard, IMO.

> There's no reason for this.

There is at least one, and I'm repeating myself here: parsing JSON is hard in shell without `jq`, which is a shame if all I needed was implementing `--wait-until-ready` reliably in shell and didn't care about actual information in the payload.

Reply:
> There's no reason for this.

Another reason is that if you want to use the nice deserializaion into `struct`-s API in Go or Rust, you need to know the message shape beforehand.

Reply:
Fair enough, lgtm